src/checksum/mod.rs

```diff
@@ -8,7 +9,7 @@ use std::str::FromStr;
 // is commutative (it is not sensitive to the order of adds/removes)
 // so two maps with the same contents will have the same checksum no
 // matter in what order the keys were added (or removed/replaced).
-#[derive(PartialEq)]
+#[derive(Clone, Copy, PartialEq)]
```

Comment:
I do not think that you need Clone. Remove this and remove all the calls to clone() in this PR. The Copy trait should change those assignments into copies automatically.